### ISSUE TITLE
Suggest to add the ability to specify the client IP for communication.

### ISSUE THREAD

**Issue by torvn77** [irungentoo/toxcore#1453]
yandex translate:
It so happens that through Facebook there is a connection, but no communication between clients tox.
And through chat Facebook you can exchange IP addresses. 
In this regard, I propose to add the ability to specify the ip for communication.

Original text:
Бывает так, что через фацебоок связь есть, но нет связи между клиентами tox.
И через работающий чат Фацебоок можно обменяться IP адресами. 
В связи с этим я предлагаю добавить возможность прямо указывать ip для установления связи.

**Comment by GrayHatter** [comment]
What benefit would this give you?

Google Translate 

Какая польза будет это даст вам ?

**Comment by ProMcTagonist** [comment]
> It so happens that through Facebook there is a connection, but no communication between clients tox.

It seems that he's in some highly censored network where Tox clients are unable to find each other in the DHT, but can still connect directly if supplied with the correct IP?

**Comment by JacobHenner** [comment]
> It seems that he's in some highly censored network where Tox clients are unable to find each other in the DHT, but can still connect directly if supplied with the correct IP?

This could be useful, especially in situations where the hosts used for DHT bootstrapping are censored.

**Comment by torvn77** [comment]
>It seems that he's in some highly censored network where Tox clients are unable to find each other in the DHT, but can still connect directly if supplied with the correct IP?

Yes, this is what I think too.
But the reason for making me the proposal was the need for way to establish the link when for any reason, such as unstable network connection or insufficient computer speed will not create a stable channel audio.

Original text:
Да, об этом я думаю тоже.
Но поводом для внесения мной предложения стала потребность в способе установить связь когда по каким либо причинам, например нестабильное соединение с сетью или недостаточное быстродействие компьютера не позволяют создать стабильный канал звуковой связи.

**Comment by TheNain38** [comment]
Yes it could be useful, so even if you can't access the bootstrapping nodes you could connect to someone.

**Comment by aaannndddyyy** [comment]
iirc, you can already do a manual bootstrap

**Comment by gaffatape** [comment]
I think this is a direction that would solve a lot of other problems as well. If the toxcore API was being modularised even more to separate the DHT from the communication layer.

* Toxcore provides a direct communication IP to IP.
* Toxcore ALSO provides a means to find the IP of friends by a DHT.

I don't know if toxcore tries to detect the online status of friends through the DHT or if it does it with a direct connection attempt to the IP, but the latter would be the best solution when the DHT become large and would make it simpler to seperate DHT and IP-to-IP communication in the library.

The mobile clients have problems with participating in a DHT, so they would benefit with an option of not having to do that and if you're just a few friends on a network you wouldn't need a DHT either. DHTs are great, but they do take up a lot of bandwith to sustain themselves.

I can also see future possibilities in being able to participate in DHTs limited to you and your friends only.

**Comment by torvn77** [comment]
No, what I want is just the ability to force the establishment of a connection with a particular client if, for some reason, the usual ways of working with clients do not allow them to find each other using DHT.
